### ISSUE TITLE
Git thinks files are changed, while they are not

### ISSUE THREAD

**Issue by Maxim-Mazurok** [denoland/deno#1995]
I'm getting following `git status` output in the clean repository:
```
$ git status
# On branch master
# Changes not staged for commit:
#
#       modified:   website/images/deno_logo.png
#       modified:   website/images/schematic_v0.2.png
#
```
`diff` shows:
```
$ git diff website/images/deno_logo.png
warning: CRLF will be replaced by LF in website/images/deno_logo.png.
The file will have its original line endings in your working directory.
diff --git a/website/images/deno_logo.png b/website/images/deno_logo.png
index 24f6251..7bf0ef3 100644
Binary files a/website/images/deno_logo.png and b/website/images/deno_logo.png differ
```
I've tried resetting, cloning again and again in new directories with no luck.

---

Finally, I've found the root of the problem:
https://github.com/denoland/deno/blob/efe9c18b459dbec71b213e8dc87e2e1b02477139/.gitattributes#L2
Here, we have `text=auto`, which means that git will try to figure out by itself which files are text out of `*` (every file).
And git thinks that those `.png` files are text and attempt to convert them.

I have `git --version` 1.8.3.1 on CentOS 7.6.1810.

---

Solutions:

1. Add 
```*.png -text```
line after https://github.com/denoland/deno/blob/efe9c18b459dbec71b213e8dc87e2e1b02477139/.gitattributes#L2
2. Update git version. Building the most recent version of git from sources (2.21.0.196.g041f5ea.dirty) fixed this issue for me.

We probably might want to open PR with the first solution, because git 1.8 is still the latest in official CentOS base updates repository. So, other developers might face similar issue. It also happens in submodules.

**Comment by ry** [comment]
I haven't experienced this, so if the first solution fixes your problem please open a PR for it.

**Comment by methodbox** [comment]
> I'm getting following `git status` output in the clean repository:
> 
> ```
> $ git status
> # On branch master
> # Changes not staged for commit:
> #
> #       modified:   website/images/deno_logo.png
> #       modified:   website/images/schematic_v0.2.png
> #
> ```
> `diff` shows:
> 
> ```
> $ git diff website/images/deno_logo.png
> warning: CRLF will be replaced by LF in website/images/deno_logo.png.
> The file will have its original line endings in your working directory.
> diff --git a/website/images/deno_logo.png b/website/images/deno_logo.png
> index 24f6251..7bf0ef3 100644
> Binary files a/website/images/deno_logo.png and b/website/images/deno_logo.png differ
> ```
> I've tried resetting, cloning again and again in new directories with no luck.
> 
> Finally, I've found the root of the problem:
> [deno/.gitattributes](https://github.com/denoland/deno/blob/efe9c18b459dbec71b213e8dc87e2e1b02477139/.gitattributes#L2)
> 
> Line 2 in [efe9c18](/denoland/deno/commit/efe9c18b459dbec71b213e8dc87e2e1b02477139)
> 
>  * text=auto eol=lf 
> 
> Here, we have `text=auto`, which means that git will try to figure out by itself which files are text out of `*` (every file).
> And git thinks that those `.png` files are text and attempt to convert them.
> I have `git --version` 1.8.3.1 on CentOS 7.6.1810.
> 
> Solutions:
> 
> 1. Add
>    `*.png -text`
>    line after 
>      
>        
>          [deno/.gitattributes](https://github.com/denoland/deno/blob/efe9c18b459dbec71b213e8dc87e2e1b02477139/.gitattributes#L2)
>        
>        
>             Line 2
>          in
>          [efe9c18](/denoland/deno/commit/efe9c18b459dbec71b213e8dc87e2e1b02477139)
>        
>        
>        
>        
>    
>            
>              
>               * text=auto eol=lf
> 2. Update git version. Building the most recent version of git from sources (2.21.0.196.g041f5ea.dirty) fixed this issue for me.
> 
> We probably might want to open PR with the first solution, because git 1.8 is still the latest in official CentOS base updates repository. So, other developers might face similar issue. It also happens in submodules.

Your CRLF warning makes me think either you're working from Windows or moved those files from a Windows machine.

Did you download them in Windows first?

Try a wget or curl 'https://url/filename' -o filename.zip on the remote machine if so. This will stop Windows from writing the CRLF line endings.

Alernatively, you can configure your local shell to prevent it and download it with that if you're extracting files locally first for some reason.

**Comment by Maxim-Mazurok** [comment]
No, @methodbox, I'm not on Windows machine and I haven't moved those files from Windows machine.
It's a bug in an old git version and it was fixed by using the latest git version.
The problem was with `.png` files, which are binary files and shouldn't be treated by git as files that have LF/CRLF line endings. It just makes no sense. So, it was clearly a git issue.
Let me know if you think otherwise, I will experiment with it.

**Comment by methodbox** [comment]
> No, @methodbox, I'm not on Windows machine and I haven't moved those files from Windows machine.
> It's a bug in an old git version and it was fixed by using the latest git version.
> The problem was with `.png` files, which are binary files and shouldn't be treated by git as files that have LF/CRLF line endings. It just makes no sense. So, it was clearly a git issue.
> Let me know if you think otherwise, I will experiment with it.

Sorry, I missed the `png` part somehow. Yes, that's odd. Sorry, just trying to help!

**Comment by ry** [comment]
Cool. So the solution is to upgrade git. Closing this issue.

**Comment by Maxim-Mazurok** [comment]
Yes, @ry, it's one of the solutions.
The other solution (for those with the old git version, but latest available in official CentOS repo) is here: #2161
